src/main/java/org/sonarlint/intellij/SonarApplication.java

```diff
@@ -68,4 +75,19 @@ public void disposeComponent() {
   public String getComponentName() {
     return getClass().getSimpleName();
   }
+
+  private static void cleanOldWorkDir() {
+    Path oldWorkDir = Paths.get(PathManager.getConfigPath()).resolve("sonarlint").resolve("work");
+    if (!Files.isDirectory(oldWorkDir)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this with a call to the "toFile().isDirectory()" method [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3725)